From 64b175407eaeb8578bb142ad4db796df0546e526 Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Tue, 23 Oct 2018 16:32:34 +0800
Subject: [PATCH 4950/5242] MLK-20099-1 ARM64: dts: use external osc as refclk
 on 8mm evk board

commit  435a64d39ebef392cd119bba9082ed87d22c0b0c from
https://source.codeaurora.org/external/imx/linux-imx.git

Use the external osc as the pcie refclk on 8mm evk
rev c board.
Create another -revb.dts for the back compatible usage.
Because that internal pll is used as pcie ref clock on
the rev b board in default.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    3 +-
 .../boot/dts/freescale/fsl-imx8mm-evk-revb.dts     |   32 ++++++++++++++++++++
 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts   |    2 +-
 3 files changed, 35 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-revb.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 5df185c..1a671f8 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -121,4 +121,5 @@ dtb-$(CONFIG_ARCH_FSL_IMX8MM) += fsl-imx8mm-evk.dtb \
 				 fsl-imx8mm-ddr4-val.dtb \
 				 fsl-imx8mm-evk-rm67191.dtb \
 				 fsl-imx8mm-evk-root.dtb \
-				 fsl-imx8mm-evk-inmate.dtb
+				 fsl-imx8mm-evk-inmate.dtb \
+				 fsl-imx8mm-evk-revb.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-revb.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-revb.dts
new file mode 100644
index 0000000..c2f89aa
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-revb.dts
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "fsl-imx8mm-evk.dts"
+
+/ {
+	model = "FSL i.MX8MM EVK RevB board";
+};
+
+/*
+ * External OSC is used as PCIe REFCLK on RevC board.
+ * Use the -revb.dts file to distiguish the different
+ * HW design.
+ */
+&pcie0{
+	ext_osc = <0>;
+	status = "okay";
+};
+
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
index 43ae138..8042923 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
@@ -926,7 +926,7 @@
 	pinctrl-0 = <&pinctrl_pcie0>;
 	disable-gpio = <&gpio1 5 GPIO_ACTIVE_LOW>;
 	reset-gpio = <&gpio4 21 GPIO_ACTIVE_LOW>;
-	ext_osc = <0>;
+	ext_osc = <1>;
 	status = "okay";
 };
 
-- 
1.7.9.5

